### PR TITLE
move module to viam-modules

### PR DESCRIPTION
**Pull request by martha-johnston**
updated the GitHub link in the meta.json to point to the repo that is now in the `Viam-modules` org. built the module locally and tested on a pi, no issues

### CODE REVIEW

**Comment by martha-johnston** [comment]
any thoughts on why this check would be failing?

**Comment by jeremyrhyde** [comment]
The `github_token` is missing: 
https://github.com/viam-modules/rplidar/blob/033302c110c0df3c54364e55f5af2c258ec53f34/.github/workflows/pr-labeler.yml#L20. 

This is most likely due to the need to update the workflows to reference the new organization. If you look through the workflows (see pr-labler.yml and main.yml as examples). They reference `viamrobotics` instead of the new `viam-modules`. I would try updating every instances of these and see if that fixes the issue.

Note: Once updated, the updates will not be used in the PR until they are already in main

**Comment by martha-johnston** [comment]
> Note: Once updated, the updates will not be used in the PR until they are already in main

does this mean I have to bypass branch protections and merge with the checks failing?

**Comment by jeremyrhyde** [comment]
> > Note: Once updated, the updates will not be used in the PR until they are already in main
> 
> does this mean I have to bypass branch protections and merge with the checks failing?

Yes, there is a way around that by relinking which workflows are run to the one in the current PR. This process looks like this:
1. Create a branch for your workflow changes
2. Create a PR updating the workflows to reference workflows in this PR and not main (Note: this will run on main workflows)
3. Create a PR from your branch to main (Note: this will run on branch workflows)
4. Merge this PR once all changes have been made to the workflows and tests are passing
5. Create a new PR reverting back to using main PR
6. Delete branch

However, since you are actively solving this process and this process doesn't affect the actual code itself it is usually safe to merge without checking for a few PRs while you get it sorted.

@randhid to confirm as this may dilute the commit history with several PRs in sequence

**Comment by randhid** [comment]
@martha-johnston this is a cgo module that makes an makes an appimage and will need to be updated to use the new style of workflow that the C++ modules use. The slack thread that discusses this is : https://viaminc.slack.com/archives/C02P30REXFW/p1724767442860879, We're trying to move away from the older credentials.

File: .canon.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 rplidar:
-    image_amd64: ghcr.io/viamrobotics/canon:amd64
-    image_arm64: ghcr.io/viamrobotics/canon:arm64
+    image_amd64: ghcr.io/viam-modules/canon:amd64
+    image_arm64: ghcr.io/viam-modules/canon:arm64
```
Comment (Otterverse) [inline .canon.yaml:3]: FYI, image addresses haven't moved, so all the changes like this throughout that are pointing to canon shouldn't be changed.

Also, the old "canon" image is way out of date now. If it still works, that's fine. But the maintained one (from RDK) is `ghcr.io/viamrobotics/rdk-devenv`

Comment (randhid) [reply .canon.yaml:3]: great catch, thanks james!

**Comment by martha-johnston** [comment]
> @martha-johnston this is a cgo module that makes an makes an appimage and will need to be updated to use the new style of workflow that the C++ modules use. The slack thread that discusses this is : https://viaminc.slack.com/archives/C02P30REXFW/p1724767442860879, We're trying to move away from the older credentials.

@randhid I see the thread but it's not clear to me what the next steps are... am I supposed to be building an AppImage manually or getting new creds from someone or something else?

File: .canon.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 rplidar:
-    image_amd64: ghcr.io/viamrobotics/canon:amd64
-    image_arm64: ghcr.io/viamrobotics/canon:arm64
+    image_amd64: ghcr.io/viamrobotics/rdk-devenv/canon:amd64
+    image_arm64: ghcr.io/viamrobotics/rdk-devenv/canon:arm64
```
Comment (Otterverse) [inline .canon.yaml:3]:
```suggestion
    image_amd64: ghcr.io/viamrobotics/rdk-devenv:amd64
    image_arm64: ghcr.io/viamrobotics/rdk-devenv:arm64
```
Same as the file in RDK: https://github.com/viamrobotics/rdk/blob/main/.canon.yaml#L4-L5

**Comment by randhid** [comment]
> > @martha-johnston this is a cgo module that makes an makes an appimage and will need to be updated to use the new style of workflow that the C++ modules use. The slack thread that discusses this is : https://viaminc.slack.com/archives/C02P30REXFW/p1724767442860879, We're trying to move away from the older credentials.
> 
> @randhid I see the thread but it's not clear to me what the next steps are... am I supposed to be building an AppImage manually or getting new creds from someone or something else?

You can follow what was done to the intel realsense [here](https://github.com/viam-modules/viam-camera-realsense/commits/main/) since that workflow also uses an app image. This repo's credentials do not match up with the upload step. You can ask etai for further explanations, maybe he can write up something for us all to use for viam-modules since he's the one who wants this change, and that would be helpful to us all.

File: .canon.yaml (hunk 2 above)
Comment (martha-johnston) [reply .canon.yaml:3]: ah I see my bad

Comment (martha-johnston) [reply .canon.yaml]: should (hopefully) be all fixed now!

File: .canon.yaml (hunk 1 above)
Comment (martha-johnston) [reply .canon.yaml]: cc @hexbabe I was just looking at the real sense repo and I noticed that you also changed the canon addresses to `viam-modules`, so you might need to change them. (apologies if the real sense doesn't use the same canon images and this is irrelevant but thought I'd double check)

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -1,5 +1,9 @@
 name: Build and Publish Latest
 
+permissions:
```
Comment (etai-shuchatowitz) [inline .github/workflows/main.yml:3]: If you're not changing the `auth` -- you probably don't need this (I don't think this uses GCP)

File: .github/workflows/pullrequest-trusted.yml [hunk 4]
```diff
@@ -1,6 +1,10 @@
 
 name: Pull Request Update
 
+permissions:
```
Comment (etai-shuchatowitz) [inline .github/workflows/pullrequest-trusted.yml:4]: Ditto to above

**Comment by martha-johnston** [comment]
@randhid @jeremyrhyde I worked with Etai to update the workflow process, but it seems unrelated to the failing check and the `PR_TOKEN` issue. is it still true that these updates won't be used in this PR, but if I force merge this it will be correct for future PRs?

**Comment by randhid** [comment]
I am okay with force merging this and diluting with a few commits.

**Comment by martha-johnston** [comment]
great, I'll do that now.

**Comment by martha-johnston** [comment]
> Update: The github_token issue is related to the need to access the PR_TOKEN secret (see https://viam.atlassian.net/wiki/spaces/ENG/pages/196698/Dev+Ops for more details). We can decide to add a PR_token to the new org OR change to using the autogenerated GITHUB_TOKEN (https://docs.github.com/en/actions/security-for-github-actions/security-guides/automatic-token-authentication)

actually, @jeremyrhyde I want to clarify this. do I need to generate a new `PR_TOKEN` for the `viam-modules` org? or is changing the `viamrobotics` references to `viam-modules` sufficient? If the answer is the former, does anyone have access to the org settings that could help me out with that? @randhid

**Comment by martha-johnston** [comment]
okay deleted the pr-labeler workflow and now am going to force merge since this will continue to fail against main

File: .github/workflows/deploy_module.yml [hunk 5]
```diff
@@ -72,7 +72,7 @@ jobs:
         name: appimage-${{ matrix.platform }}
     - id: stripped-version
       run: echo version=${VERSION#v} >> $GITHUB_OUTPUT
-    - uses: viamrobotics/upload-module@main
+    - uses: viam-modules/upload-module@main
```
Comment (JohnN193) [inline .github/workflows/deploy_module.yml:75]: i think this change needs to be reverted? or at least I didn't change this for cartographer & my module uploaded

Comment (martha-johnston) [reply .github/workflows/deploy_module.yml]: done! going to merge now

**Pull request merged**